sdk/src/Core/Amazon.Runtime/Credentials/AssumeRoleAWSCredentials.cs

```diff
@@ -87,34 +89,49 @@ public AssumeRoleAWSCredentials(AWSCredentials sourceCredentials, string roleArn
             PreemptExpiryTime = TimeSpan.FromMinutes(15);
         }
 
+#if NET6_0_OR_GREATER
+        [System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", 
+            Justification = "Reflection code is only used as a fallback in case the SDK was not trimmed. Trimmed scenarios should register dependencies with Amazon.RuntimeDependencyRegistry.GlobalRuntimeDependencyRegistry")]
+#endif
         protected override CredentialsRefreshState GenerateNewCredentials()
         {
             var region = FallbackRegionFactory.GetRegionEndpoint() ?? DefaultSTSClientRegion;
-            ICoreAmazonSTS coreSTSClient;
+            ICoreAmazonSTS coreSTSClient = GlobalRuntimeDependencyRegistry.Instance.GetInstance<ICoreAmazonSTS>(ServiceClientHelpers.STS_ASSEMBLY_NAME, ServiceClientHelpers.STS_SERVICE_CLASS_NAME, 
```

Comment:
This still does not support the Options.ProxySettings when coming from the global runtime unless a dev sets them for the obvious reason the instance is already constructed. How is a dev going to know to set these values when they may have been using Options before? Documentation? Prior comment: https://github.com/aws/aws-sdk-net/pull/3045#discussion_r1320364108

Would `Options` need to be passed in the `SecurityTokenServiceClientContext`?

Reply:
I added the proxy settings to the context. I don't think adding the whole `Options` class should be done. There is different type of options for each possible action and the other properties are more for making the assume call not constructing the STS client.